### PR TITLE
refactor: 💡 size and frame reader

### PR DESCRIPTION
**Pull request by xiaoyu0722**

### CODE REVIEW

File: Sources/FioriSwiftUICore/Experimental/Popover/View+Extension.swift [hunk 1]
```diff
@@ -10,40 +10,18 @@ extension View {
...
-                let frame = geometry.frame(in: coordinateSpace)
-                Color.clear
-                    .preference(key: ContentFrameReaderPreferenceKey.self, value: frame)
-                    .onPreferenceChange(ContentFrameReaderPreferenceKey.self) { newValue in
-                        rect(newValue)
-                    }
-            }
-            .hidden()
-        )
+        onGeometryChange(for: CGRect.self) { proxy in
```
Comment (billzhou0223) [inline Sources/FioriSwiftUICore/Experimental/Popover/View+Extension.swift:13]: This is a iOS 18 only API. But we still need to support iOS 17.

**Review by billzhou0223 — CHANGES_REQUESTED** [review]

File: Sources/FioriSwiftUICore/Experimental/Popover/View+Extension.swift (hunk 1 above)
Comment (billzhou0223) [reply Sources/FioriSwiftUICore/Experimental/Popover/View+Extension.swift:13]: Found this API works in iOS 17. Close this comment

**Review by billzhou0223 — APPROVED** [review]

**Comment by billzhou0223** [comment]
@xiaoyu0722 Is this PR complete? If so please remove the `do not merge` tag.

**Comment by xiaoyu0722** [comment]
`onGeometryChange` needs swift 6.0. So I will close this PR and reopen it when ci tool is update to Xcode 16.

**Pull request closed without merging**